tempesta_fw/sock.c

```diff
@@ -1049,7 +1130,14 @@ ss_connect(struct sock *sk, struct sockaddr *uaddr, int uaddr_len, int flags)
 	if (sk->sk_state != TCP_CLOSE)
 		return -EISCONN;
 
-	return sk->sk_prot->connect(sk, uaddr, uaddr_len);
+	if (ss_active_guard_enter(SS_SOFTIRQ_VAL))
```

Comment:
Logically you're right and I change the constant. However, technically there is no difference. I used LISTEN constant just to get a barrier which guarantees that after closing listening socket no new connection appears and we can safely traverse all established connections and close them all. So at this point the only guarding, which prevents calling the callback on shutdown phase, has sense and exact constant value doesn't matter.